include/GooseEPM/System.h

```diff
@@ -334,6 +334,24 @@ class SystemAthermal {
         return m_propagator.shape();
     }
 
+    /**
+     * @brief Set the propagator.
+     * @param propagator Propagator.
+     */
+    void set_propagator(const array_type::tensor<double, 2>& propagator)
+    {
+        m_propagator = propagator;
+    }
```

Comment:
```suggestion
     * @param value Propagator.
     */
    void set_propagator(const array_type::tensor<double, 2>& value)
    {
        GOOSEEPM_ASSERT(xt::has_shape(value, m_propagator.shape()), std::out_of_range);
        m_propagator = value;
    }
```

Reply:
I'm not sure to understand why we're having problems with the argument name in this case, but not for other cases like set_sigmay or set_sigma